crates/bevy_window/src/window.rs

```diff
@@ -542,8 +542,78 @@ impl WindowResizeConstraints {
     }
 }
 
+/// Cursor icon.
+#[derive(Debug, Clone, Reflect, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serialize",
+    derive(serde::Serialize, serde::Deserialize),
+    reflect(Serialize, Deserialize)
+)]
+#[reflect(Debug, Default)]
+pub enum CursorIcon {
+    /// Custom cursor image data.
+    Custom(CustomCursor),
+    /// Native cursor icon.
+    Native(NativeCursorIcon),
+}
+
+impl Default for CursorIcon {
+    fn default() -> Self {
+        CursorIcon::Native(NativeCursorIcon::Default)
+    }
+}
+
+impl From<NativeCursorIcon> for CursorIcon {
+    fn from(icon: NativeCursorIcon) -> Self {
+        CursorIcon::Native(icon)
+    }
+}
+
+impl From<CustomCursor> for CursorIcon {
+    fn from(cursor: CustomCursor) -> Self {
+        CursorIcon::Custom(cursor)
+    }
+}
+
+/// Custom cursor image data.
+#[derive(Debug, Clone, Reflect, PartialEq, Eq, Hash)]
+#[cfg_attr(
+    feature = "serialize",
+    derive(serde::Serialize, serde::Deserialize),
+    reflect(Serialize, Deserialize)
+)]
+pub enum CustomCursor {
+    /// Image data in RGBA format. Cursor creation can fail if the image data is invalid.
+    /// Rgba data must be a multiple of 4 bytes in length. Width times height must match
+    /// the length of the data / 4. The hotspot must be within the image bounds.
+    /// `image_to_rgba_pixels` can be used to convert an `Image` to RGBA pixels.
+    Image {
```

Comment:
Man, this is really rough as an API. Totally agree that a `Handle<Image>` would be much better.